### ISSUE TITLE
metaviralSPAdes Installation issues

### ISSUE THREAD

**Issue by Adigorla** [ablab/spades#618]
Hi, Im trying to install metaviralSPAdes but cant pull from the branch "because you have unmerged files". I also run into the same issue when trying to pull viralVerify. Do you have any recommendations on how to fix this or alternative methods to install?

Also, in the Issues threads discussions I've heard some mentions of "rnaviralSPAdes". Can you provide some information on the correct context to use this tool and how to install it? Thanks!

**Comment by asl** [comment]
rnaviralSPAdes is a part of coronaSPAdes. Please refer to https://cab.spbu.ru/software/coronaspades/ for instructions

The issues you're seeing when trying to pull the branch are likely due to some local problems, either with git or with the system, I'm afraid we cannot help with them.

**Comment by Adigorla** [comment]
Hi, So I'm having new issues with metaviralSPAdes installation. Im getting the following errors while trying to source build. I believe I have all necessary libraries/tools installed. Could you please help fix this issue? 

```
[ 60%] Linking CXX static library libConsensusCore.a
[ 60%] Built target ConsensusCore
[ 60%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/pipeline/extenders_logic.cpp.o
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/extension_chooser2015.cpp.o
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph.cpp.o
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph_constructor.cpp.o
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph_visualizer.cpp.o
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/connection_condition2015.cpp.o
[ 61%] Linking CXX static library libinput.a
[ 61%] Built target input
[ 61%] Building CXX object common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/path_polisher.cpp.o
In file included from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/graph_support/contig_output.hpp:10:0,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/paths/bidirectional_path_io/io_support.hpp:10,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/paths/bidirectional_path_io/bidirectional_path_output.hpp:13,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/pipeline/launcher.hpp:14,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/pipeline/launcher.cpp:5:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/stats/picture_dump.hpp: In instantiation of ‘size_t debruijn_graph::stats::Nx(Graph&, double) [with Graph = debruijn_graph::DeBruijnGraph; size_t = long unsigned int]’:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/pipeline/launcher.cpp:168:103:   required from here
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/stats/picture_dump.hpp:448:24: warning: conversion to ‘double’ from ‘__gnu_cxx::__alloc_traits<std::allocator<long unsigned int> >::value_type {aka long unsigned int}’ may alter its value [-Wconversion]
         if (lengths[i] >= len_perc)
                         
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/extension_chooser2015.cpp: In member function ‘path_extend::ExtensionChooser::EdgeContainer path_extend::ExtensionChooser2015::FindNextUniqueEdge(path_extend::EdgeId) const’:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/extension_chooser2015.cpp:29:19: warning: conversion to ‘double’ from ‘long unsigned int’ may alter its value [-Wconversion]
         if (sum < absolute_weight_threshold_) {
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
c++: internal compiler error: Bus error (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://gcc.gnu.org/bugs.html> for instructions.
make[2]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph_visualizer.cpp.o] Error 4
make[2]: *** Waiting for unfinished jobs....
c++: internal compiler error: Bus error (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://gcc.gnu.org/bugs.html> for instructions.
make[2]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/extension_chooser2015.cpp.o] Error 4
In file included from /gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/move.h:57:0,
                 from /gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_pair.h:59,
                 from /gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_algobase.h:64,
                 from /gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/char_traits.h:39,
                 from /gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/string:40,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/utils/perf/perfcounter.hpp:10,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/utils/logger/logger.hpp:9,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/core/observable_graph.hpp:10,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/core/graph.hpp:10,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/paths/bidirectional_path.hpp:17,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/weight_counter.hpp:11,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/extension_chooser.hpp:18,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/path_extender.hpp:16,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/pipeline/extenders_logic.hpp:7,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/pipeline/extenders_logic.cpp:5:
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/type_traits: In instantiation of ‘struct std::__or_<std::is_function<std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > >, std::is_array<std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > > >’:
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/type_traits:125:12:   required from ‘struct std::__or_<std::is_void<const std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >&>, std::is_function<std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > >, std::is_array<std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > > >’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/type_traits:1558:12:   required from ‘struct std::is_convertible<const std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >&, std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > >’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/type_traits:143:12:   required from ‘struct std::__and_<std::is_convertible<const std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >&, std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > >, std::is_convertible<const std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >&, std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> > > >’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_pair.h:107:43:   required from ‘static constexpr bool std::_PCC<<anonymous>, _T1, _T2>::_ImplicitlyConvertiblePair() [with _U1 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; _U2 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; bool <anonymous> = true; _T1 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; _T2 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >]’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_pair.h:239:43:   required by substitution of ‘template<class _U1, class _U2, typename std::enable_if<(_ConstructiblePair<_U1, _U2>() && _ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const _T1&, const _T2&) [with _U1 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; _U2 = std::_Rb_tree_const_iterator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; typename std::enable_if<(_ConstructiblePair<_U1, _U2>() && _ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = <missing>]’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_tree.h:1764:15:   required from ‘std::pair<std::_Rb_tree_const_iterator<_Val>, std::_Rb_tree_const_iterator<_Val> > std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::equal_range(const _Key&) const [with _Key = omnigraph::impl::EdgeId; _Val = std::pair<const omnigraph::impl::EdgeId, long unsigned int>; _KeyOfValue = std::_Select1st<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; _Compare = std::less<omnigraph::impl::EdgeId>; _Alloc = std::allocator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >]’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_tree.h:2346:61:   required from ‘std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::size_type std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::count(const _Key&) const [with _Key = omnigraph::impl::EdgeId; _Val = std::pair<const omnigraph::impl::EdgeId, long unsigned int>; _KeyOfValue = std::_Select1st<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; _Compare = std::less<omnigraph::impl::EdgeId>; _Alloc = std::allocator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::size_type = long unsigned int]’
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/bits/stl_multimap.h:795:30:   required from ‘std::multimap<_Key, _Tp, _Compare, _Alloc>::size_type std::multimap<_Key, _Tp, _Compare, _Alloc>::count(const key_type&) const [with _Key = omnigraph::impl::EdgeId; _Tp = long unsigned int; _Compare = std::less<omnigraph::impl::EdgeId>; _Alloc = std::allocator<std::pair<const omnigraph::impl::EdgeId, long unsigned int> >; std::multimap<_Key, _Tp, _Compare, _Alloc>::size_type = long unsigned int; std::multimap<_Key, _Tp, _Compare, _Alloc>::key_type = omnigraph::impl::EdgeId]’
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/genome_consistance_checker.hpp:51:34:   required from here
/gstore/apps/GCCcore/6.3.0/include/c++/6.3.0/type_traits:120:12: internal compiler error: Bus error
     struct __or_<_B1, _B2>
            ^~~~~~~~~~~~~~~
In file included from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/scaffold_graph.hpp:6:0,
                 from /gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/scaffold_graph.cpp:1:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/graph_support/genomic_quality.hpp: In destructor ‘virtual debruijn_graph::QualityLoggingRemovalHandler<Graph>::~QualityLoggingRemovalHandler()’:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/utils/logger/logger.hpp:123:12: internal compiler error: Bus error
       std::cerr << ". Message="  <<  msg << "\n";                       \
            ^
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/utils/logger/logger.hpp:137:46: note: in expansion of macro ‘LOG_MSG’
 # define TRACE(message)                      LOG_MSG(logging::L_TRACE, message)
                                              ^~~~~~~
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/assembly_graph/graph_support/genomic_quality.hpp:204:9: note: in expansion of macro ‘TRACE’
         TRACE("Overall stats: total removed = " << total_
         ^~~~~
c++: internal compiler error: Bus error (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://gcc.gnu.org/bugs.html> for instructions.
make[2]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph_constructor.cpp.o] Error 4
0xb0393f crash_signal
	../../gcc/toplev.c:333
0x1185840 IS_ADHOC_LOC(unsigned int)
	../../libcpp/include/line-map.h:976
0x1185840 linemap_location_in_system_header_p(line_maps*, unsigned int)
	../../libcpp/line-map.c:1179
0x67489c constrain_class_visibility(tree_node*)
	../../gcc/cp/decl2.c:2552
0xb0393f crash_signal
	../../gcc/toplev.c:333
0x66ecfe finish_struct_1(tree_node*)
	../../gcc/cp/class.c:7231
0x656a6b instantiate_class_template_1
	../../gcc/cp/pt.c:10337
0x656a6b instantiate_class_template(tree_node*)
	../../gcc/cp/pt.c:10377
0x69c980 cp_parser_binary_expression
	../../gcc/cp/parser.c:8763
0x69d100 cp_parser_assignment_expression
	../../gcc/cp/parser.c:9063
0x69f39a cp_parser_expression
	../../gcc/cp/parser.c:9232
0x69f8ef cp_parser_expression_statement
	../../gcc/cp/parser.c:10695
0x6ab52f cp_parser_statement
	../../gcc/cp/parser.c:10546
0x6abd94 cp_parser_statement_seq_opt
	../../gcc/cp/parser.c:10818
0x6abe6f cp_parser_compound_statement
	../../gcc/cp/parser.c:10772
0x6b1045 cp_parser_implicitly_scoped_statement
	../../gcc/cp/parser.c:11935
0x6ab903 cp_parser_selection_statement
	../../gcc/cp/parser.c:10900
0x6ab903 cp_parser_statement
	../../gcc/cp/parser.c:10401
0x6abd94 cp_parser_statement_seq_opt
	../../gcc/cp/parser.c:10818
0x6abe6f cp_parser_compound_statement
	../../gcc/cp/parser.c:10772
0x6b1045 cp_parser_implicitly_scoped_statement
	../../gcc/cp/parser.c:11935
0x6b1f3c cp_parser_iteration_statement
	../../gcc/cp/parser.c:11621
0x6aad81 cp_parser_statement
	../../gcc/cp/parser.c:10407
0x6abd94 cp_parser_statement_seq_opt
	../../gcc/cp/parser.c:10818
0x6abe6f cp_parser_compound_statement
	../../gcc/cp/parser.c:10772
0x6abf83 cp_parser_function_body
	../../gcc/cp/parser.c:20717
0x6abf83 cp_parser_ctor_initializer_opt_and_function_body
	../../gcc/cp/parser.c:20753
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <http://gcc.gnu.org/bugs.html> for instructions.
0x6b7e3d complete_type(tree_node*)
	../../gcc/cp/typeck.c:131
0x6b7edf complete_type_or_maybe_complain(tree_node*, tree_node*, int)
	../../gcc/cp/typeck.c:143
0x620489 xref_basetypes(tree_node*, tree_node*)
	../../gcc/cp/decl.c:12933
0x656455 instantiate_class_template_1
	../../gcc/cp/pt.c:9966
0x656455 instantiate_class_template(tree_node*)
	../../gcc/cp/pt.c:10377
0x6b7e3d complete_type(tree_node*)
	../../gcc/cp/typeck.c:131
0x6dd004 lookup_member(tree_node*, tree_node*, int, bool, int)
	../../gcc/cp/search.c:1263
0x71642d lookup_qualified_name(tree_node*, tree_node*, int, bool, bool)
	../../gcc/cp/name-lookup.c:4550
0x640ab5 tsubst_qualified_id
	../../gcc/cp/pt.c:13838
0x6421d2 tsubst_copy_and_build(tree_node*, tree_node*, int, tree_node*, bool, bool)
	../../gcc/cp/pt.c:16280
0x63cac8 tsubst_expr(tree_node*, tree_node*, int, tree_node*, bool)
	../../gcc/cp/pt.c:15876
0x6453bc tsubst_template_arg
	../../gcc/cp/pt.c:10396
0x64a682 tsubst_template_args
	../../gcc/cp/pt.c:11272
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <http://gcc.gnu.org/bugs.html> for instructions.
make[2]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/scaffolder2015/scaffold_graph.cpp.o] Error 1
make[2]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/pipeline/extenders_logic.cpp.o] Error 1
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/connection_condition2015.cpp: In member function ‘virtual path_extend::Connections path_extend::LongReadsLibConnectionCondition::ConnectedWith(path_extend::EdgeId, const path_extend::ScaffoldingUniqueEdgeStorage&) const’:
/gstore/home/gorlap/extRepos/metaviral_publication/assembler/src/common/modules/path_extend/scaffolder2015/connection_condition2015.cpp:110:28: warning: conversion to ‘double’ from ‘long unsigned int’ may alter its value [-Wconversion]
         if (iter->second < min_read_count_){
                            ^~~~~~~~~~~~~~~
make[1]: *** [common/modules/path_extend/CMakeFiles/path_extend.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by asl** [comment]
Please do not hijack the closed issues, open new issue.

The log reads:
```
c++: internal compiler error: Bus error (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://gcc.gnu.org/bugs.html> for instructions.
```

So, it seems that your C++ compiler is buggy and therefore crashed. Consider upgrading to the never version.
